argus/backends/azure/client.py

```diff
@@ -0,0 +1,240 @@
+
+        self._vm_name = "argusvm-" + str(random.randint(0,10000))
+        self._os_disk_name = "argusdisk-" + str(random.randint(0,10000))
+        self._vnet_name = "argusvnet-" + str(random.randint(0,10000))
+        self._sub_net_name = "argussubnet-" + str(random.randint(0,10000))
+        self._nic_name = "argusnic-" + str(random.randint(0,10000))
+        self._vip_name = "argusvip-" + str(random.randint(0,10000))
+        self._ip_config_name = "argusip-" + str(random.randint(0,10000))
+        self._sec_group_name = 'argussecgrp-' + str(random.randint(0,10000))
+
+    def create_vm(self):
+        nic = self.create_nic()
+        vm_parameters = self._create_vm_parameters(self._vm_name, self._vm_username,
+            self._vm_password, self._os_disk_name, nic.id, VM_REFERENCE['windows'])
+        async_vm_creation = self._compute_client.virtual_machines.create_or_update(
+            self._resource_group_name, self._vm_name, vm_parameters)
+
+        async_vm_start = self._compute_client.virtual_machines.start(
+            self._resource_group_name, self._vm_name)
+
+    def destroy_vm(self):
+         self._compute_client.virtual_machines.delete(self._resource_group_name, self._vm_name).result()
+         self._network_client.network_interfaces.delete(self._resource_group_name, self._nic_name).result()
+         self._network_client.virtual_networks.delete(self._resource_group_name, self._vnet_name).result()
+         self._network_client.network_security_groups.delete(self._resource_group_name, self._sec_group_name).result()
+         self._network_client.public_ip_addresses.delete(self._resource_group_name, self._vip_name).result()
+
+    def create_resource_group(self):
+        self._resource_client.resource_groups.create_or_update(
+            self._resource_group_name,
+            {
+                'location': self._availability_zone
+            })
+
+    def create_storage_account(self):
+        async_create = self._storage_client.storage_accounts.create(
+            self._resource_group_name,
+            self._storage_account_name,
+            {
+                'location': self._availability_zone,
+                'sku': {'name': 'standard_lrs'},
+                'kind': 'storage'
+            })
+        async_create.wait()
+
+    def create_nic(self):
+        """Create a Network Interface for a VM.
+        """
+
+        async_secgroup_creation = self._network_client.network_security_groups.create_or_update(
+            self._resource_group_name, self._sec_group_name, {'location':self._availability_zone})
+        secgroup_info = async_secgroup_creation.result()
+        
+        self._network_client.security_rules.create_or_update(self._resource_group_name, self._sec_group_name, 'secgr1', {'access':'allow', 'protocol':'Tcp', 'direction':'inbound', 'source_address_prefix':'*','destination_address_prefix':'*', 'source_port_range':'*', 'destination_port_range':'3389', 'priority':100}).wait()
```

Comment:
We can use a for loop and only change the required values from the dict instead of repeating these code blocks, also not pep8 compliant